### PR TITLE
Migrate some State for some screens from their composables to their respective ViewModels (#111)

### PR DESCRIPTION
**Pull request by chege4179**
In the HttpClientFactory.kt i changed the api key from the firebase remote config one because apparently the api calls were failing... I think its because of the changing of the package name in previous PRs but I'm not sure. 
The rest is as per the title

### CODE REVIEW

**Review by wangerekaharun — APPROVED** [review]

File: data/src/main/java/com/android254/data/network/util/HttpClientFactory.kt [hunk 1]
```diff
@@ -46,7 +46,7 @@ class HttpClientFactory @Inject constructor(private val tokenProvider: TokenProv
 
         install(DefaultRequest) {
             header(HttpHeaders.ContentType, ContentType.Application.Json)
-            header("Api-Authorization-Key", remoteFeatureToggle.getString("API_KEY"))
+            header("Api-Authorization-Key", "droidconKe-2020")
```
Comment (chepsi) [inline data/src/main/java/com/android254/data/network/util/HttpClientFactory.kt:49]: We should use the value from remote config

File: presentation/src/main/java/com/android254/presentation/speakers/SpeakerDetailsScreenViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    private val _uiState = MutableStateFlow<SpeakerDetailsScreenUiState>(SpeakerDetailsScreenUiState.Loading)
+    val uiState = _uiState.asStateFlow()
+
+    init {
+        savedStateHandle.get<Int>("speakerId")?.let{
+            viewModelScope.launch {
+                getSpeakerById(id = it)
+            }
+        }
+    }
```
Comment (chepsi) [inline presentation/src/main/java/com/android254/presentation/speakers/SpeakerDetailsScreenViewModel.kt:42]: let's avoid making calls in the init block

Comment (chege4179) [reply presentation/src/main/java/com/android254/presentation/speakers/SpeakerDetailsScreenViewModel.kt:42]: Noted

**Comment by chege4179** [comment]
@chepsi  I've made the requested changes

**Review by chepsi — APPROVED** [review]

**Pull request merged**